src/controllers/news.ts

```diff
@@ -0,0 +1,139 @@
+/* eslint-disable @typescript-eslint/type-annotation-spacing */
+/* eslint-disable eol-last */
+/* eslint-disable prettier/prettier */
+import News from '../models/news';
+import User from '../models/user';
+import {Request, Response, NextFunction} from 'express';
+import {createError} from '../utils/helpers';
+
+
+
+
+export const getNews = async (req: Request, res: Response) => {
+  try {
+    // console.log(req.query);
+    const queryObject =req.query;
+
+    const queryKeys= Object.keys(queryObject)
+    if(queryKeys.length ===0){
+      const news = await News.find({}).sort({publDate: 'desc'});
+      res.send(news)
+    }
+    if(queryObject.trending === 'true'){
+      const trendNews =await News.find({}).sort({clicks:'desc'})
+      res.send(trendNews)
+    }
+    else{
+      throw new Error('Query Not Matched')
+    }
+
+
+
+  } catch (e) {
+    res.status(500).send(e);
+  }
+};
+
+export const newsDetails =async(req : Request , res: Response)=>{
+  try {
+    const user =await User.findById(req.payload.id);
+    if(!user){
+      res.status(401).send({message:'Authentication Failed'});
+    }
+    const news= await News.findById(req.params.id);
+    if(!news){
+      throw createError(401,'Doesnot Exists','News for given id donot exists')
+    }
+    news.clicks+=1;
+    await news.save();
+    res.send(news)
+  } catch (error) {
+    res.status(500).send(error)
+}
+};
+
+export const addNews = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const user = await User.findById(req.payload.id);
+    if (user === null || user.adminOf.length === 0) {
+      throw createError(401, 'Unauthorized', 'Invalid Login Credentials');
+    }
+    const news = new News({
+      ...req.body,
+      createdBy: user._id,
+    });
+    await news.save();
+    res.send({news});
```

Comment:
You can use the createResponse function here , it helps to keep the responses consistent.
`return res.send(createResponse('News Added Successfully', news));`